code/modules/mob/living/carbon/xenomorph/castes/puppeteer/abilities_puppeteer.dm

```diff
@@ -0,0 +1,449 @@
+	)
+
+/datum/action/xeno_action/activable/puppet/can_use_ability(atom/target, silent = FALSE, override_flags)
+	. = ..()
+	var/mob/living/carbon/xenomorph/owner_xeno = owner
+	if(isclosedturf(target))
+		if(!silent)
+			target.balloon_alert(owner_xeno, "THATS A FUCKING WALL")
+		return FALSE
+
+	var/datum/action/xeno_action/activable/refurbish_husk/huskaction = owner.actions_by_path[/datum/action/xeno_action/activable/refurbish_husk]
+	if(length(huskaction.puppets) >= owner_xeno.xeno_caste.max_puppets)
+		owner_xeno.balloon_alert(owner_xeno, "cant have more than [owner_xeno.xeno_caste.max_puppets]!")
+		return FALSE
+
+	if(!owner_xeno.Adjacent(target))
+		if(!silent)
+			to_chat(owner_xeno, span_xenonotice("We need to be next to where we want to create a puppet."))
+		return FALSE
+
+	if(!.)
+		return
+	owner_xeno.face_atom(target)
+	//reverse gib here
+	owner_xeno.visible_message(span_warning("[owner_xeno] begins to vomit out biomass and skillfully sews various bits and pieces together!"))
+	if(!do_after(owner_xeno, 8 SECONDS, FALSE, target, BUSY_ICON_CLOCK, extra_checks = CALLBACK(owner_xeno, TYPE_PROC_REF(/mob, break_do_after_checks), list("health" = owner_xeno.health))))
+		//cancel effects
+		return FALSE
+	//cancel effects
+	owner_xeno.visible_message(span_warning("[owner_xeno] forms a repulsive puppet!"))
+	succeed_activate()
+
+/datum/action/xeno_action/activable/puppet/use_ability(atom/target)
+	var/turf/target_turf = get_turf(target)
+
+	var/datum/action/xeno_action/activable/refurbish_husk/huskaction = owner.actions_by_path[/datum/action/xeno_action/activable/refurbish_husk]
+	huskaction.add_puppet(new /mob/living/carbon/xenomorph/puppet(target_turf, owner))
+	add_cooldown()
+
+// ***************************************
+// *********** Organic Bomb
+// ***************************************
+/datum/action/xeno_action/activable/organic_bomb
+	name = "Organic Bomb"
+	action_icon_state = "organic_bomb"
+	desc = "Causes one of our puppets to detonate on selection, spewing acid out of the puppet's body in all directions, gibbing the puppet."
+	cooldown_timer = 30 SECONDS
+	plasma_cost = 100
+	target_flags = XABB_MOB_TARGET
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_ORGANICBOMB,
+	)
+
+/datum/action/xeno_action/activable/organic_bomb/use_ability(mob/living/victim)
+	. = ..()
+	var/datum/action/xeno_action/activable/refurbish_husk/huskaction = owner.actions_by_path[/datum/action/xeno_action/activable/refurbish_husk]
+	if(length(huskaction.puppets) <= 0)
+		owner.balloon_alert(owner, "no puppets")
+		return fail_activate()
+	if(!istype(victim, /mob/living/carbon/xenomorph/puppet) || !(victim in huskaction.puppets))
+		victim.balloon_alert(owner, "not our puppet")
+		return fail_activate()
+	if(!SEND_SIGNAL(victim, COMSIG_PUPPET_CHANGE_ORDER, PUPPET_SEEK_CLOSEST))
+		victim.balloon_alert(owner, "fail")
+		return fail_activate()
+	victim.balloon_alert(owner, "success")
```

Comment:
```suggestion
		victim.balloon_alert(owner, "fail")
		return fail_activate()
```

Reply:
why

Reply:
Spam, the player already sees from vfx when it succeeds

Reply:
ok